src/sentry/release_health/metrics.py

```diff
@@ -606,3 +601,342 @@ def extract_row_info(row: Mapping[str, Union[OrganizationId, str]]) -> ReleaseNa
+        # old and new backend, this should really go back through the
+        # release_health service instead of directly calling `self`. For now
+        # that makes the entire backend too hard to test though.
+        release_adoption = self.get_release_adoption(project_releases, environments)
+
+        rv = {}
+
+        fetch_has_health_data_releases = set()
+
+        for project_id, release in project_releases:
+            adoption_info = release_adoption.get((project_id, release)) or {}
+
+            total_sessions = rv_sessions.get((project_id, release, "init"))
+
+            total_users = rv_users.get((project_id, release, "init"))
+            has_health_data = bool(total_sessions)
+
+            # has_health_data is supposed to be irrespective of the currently
+            # selected rollup window. Therefore we need to run another query
+            # over 90d just to see if health data is available to compute
+            # has_health_data correctly.
+            if not has_health_data and summary_stats_period != "90d":
+                fetch_has_health_data_releases.add((project_id, release))
+
+            sessions_crashed = rv_sessions.get((project_id, release, "crashed"), 0)
+
+            users_crashed = rv_users.get((project_id, release, "crashed"), 0)
+
+            rv_row = rv[project_id, release] = {
+                "adoption": adoption_info.get("adoption"),
+                "sessions_adoption": adoption_info.get("sessions_adoption"),
+                "total_users_24h": adoption_info.get("users_24h"),
+                "total_project_users_24h": adoption_info.get("project_users_24h"),
+                "total_sessions_24h": adoption_info.get("sessions_24h"),
+                "total_project_sessions_24h": adoption_info.get("project_sessions_24h"),
+                "total_sessions": total_sessions,
+                "total_users": total_users,
+                "has_health_data": has_health_data,
+                "sessions_crashed": sessions_crashed,
+                "crash_free_users": (
+                    100 - users_crashed / total_users * 100 if total_users else None
+                ),
+                "crash_free_sessions": (
+                    100 - sessions_crashed / float(total_sessions) * 100 if total_sessions else None
+                ),
+                "sessions_errored": max(
+                    0,
+                    rv_errored_sessions.get((project_id, release), 0)
+                    - sessions_crashed
+                    - rv_sessions.get((project_id, release, "abnormal"), 0),
+                ),
+                **(
+                    rv_durations.get((project_id, release))
+                    or {
+                        "duration_p50": None,
+                        "duration_p90": None,
+                    }
+                ),
+            }
+
+            if health_stats_period:
+                rv_row["stats"] = {health_stats_period: health_stats_data[project_id, release]}
```

Comment:
it returns a defaultdict such that we can do this. I went with this approach because otherwise we would have to define and construct defaults in two unrelated places (see construction of `rv` in `_get_health_stats_for_overview`: `_make_stats` + its data dependencies would have to be constructed both inside and outside of that function).